packages/shared/src/components/search/SearchHistoryButton.tsx

```diff
@@ -0,0 +1,81 @@
+import React, {
+  MouseEventHandler,
+  ReactElement,
+  useMemo,
+  useState,
+} from 'react';
+import { useRouter } from 'next/router';
+import TimerIcon from '../icons/Timer';
+import { SimpleTooltip } from '../tooltips';
+import { Button, ButtonSize } from '../buttons/Button';
+import { useSearchHistory } from '../../hooks/search';
+import { ContextMenu, MenuItemProps } from '../fields/PortalMenu';
+import useContextMenu from '../../hooks/useContextMenu';
+import { getSearchUrl, searchPageUrl } from '../../graphql/search';
+
+const contextMenuId = 'search-history-input';
+
+export function SearchHistoryButton(): ReactElement {
+  const {
+    nodes,
+    result: { isLoading },
+  } = useSearchHistory({ limit: 3 });
+  const router = useRouter();
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const { onMenuClick, onHide } = useContextMenu({ id: contextMenuId });
+
+  const getMessage = () => {
+    if (isLoading) return 'Fetching your search queries';
```

Comment:
Is this something we came up with or validated in design/sab?

Reply:
Oh, I did not. Let me confirm what would they want 🦅